writeengine/wrapper/writeengine.cpp

```diff
@@ -220,6 +220,212 @@ void WriteEngineWrapper::findSmallestColumn(uint32_t& colId, ColStructList colSt
+                                ExtCPInfo* maxMin, bool canStartWithInvalidRange)
+{
+    if (!maxMin)
+    {
+        return ;
+    }
+    bool isUnsigned = false; // TODO: should change with type.
+    switch (colType)
+    {
+        case WR_UBYTE:
+        case WR_USHORT:
+        case WR_UINT:
+        case WR_ULONGLONG:
+        {
+            isUnsigned = true;
+            break;
+        }
+        default: // WR_BINARY is signed.
+            break;
+    }
+    if (!canStartWithInvalidRange)
+    {
+        // check if range is invalid, we can't update it.
+        if (maxMin->isInvalid())
+        {
+            return ;
+        }
+    }
+    size_t i;
+    for (i = 0; i < totalOldRow; i++) {
+        int64_t value = 0, oldValue = 0;
+        uint64_t uvalue = 0, oldUValue = 0;
+        int128_t bvalue = 0, oldBValue = 0;
+
+        // fetching. May not assign value or oldValue variables when array is not present.
+        switch (colType)
+        {
+            case WR_BYTE:
+            {
+                fetchNewOldValues<int64_t, int8_t>(value, oldValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_UBYTE:
+            {
+                fetchNewOldValues<uint64_t, uint8_t>(uvalue, oldUValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_SHORT:
+            {
+                fetchNewOldValues<int64_t, int16_t>(value, oldValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_USHORT:
+            {
+                fetchNewOldValues<uint64_t, uint16_t>(uvalue, oldUValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_MEDINT:
+            case WR_INT:
+            {
+                fetchNewOldValues<int64_t, int>(value, oldValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_UMEDINT:
+            case WR_UINT:
+            {
+                fetchNewOldValues<uint64_t, unsigned int>(uvalue, oldUValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_LONGLONG:
+            {
+                fetchNewOldValues<int64_t, int64_t>(value, oldValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_ULONGLONG:
+            {
+                fetchNewOldValues<uint64_t, uint64_t>(uvalue, oldUValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            case WR_BINARY:
+            {
+                fetchNewOldValues<int128_t, int128_t>(bvalue, oldBValue, valArrayVoid, oldValArrayVoid, i, totalNewRow);
+                break;
+            }
+            default:
+                idbassert(0 && "unknown WR type tag");
```

Comment:
If you want a new error message you should add it accordingly. Now this looks like a hack to log a specific message.